[TASK] Overhaul Startpage and menu / toctree configuration (#371)
TYPO3-Documentation/TYPO3CMS-Guide-HowToDocument: Documentation/GeneralConventions/FileStructure.rst

.. _index-rst:
.. _start-file:

Start page: :file:`Documentation/Index.rst`
-------------------------------------------

> The documentation index file at :file:`Documentation/Index.rst` is the starting

Author: ```suggestion
The documentation start page (:file:`Documentation/Index.rst`) is the entry
```

.. _index-rst:
.. _start-file:

Start page: :file:`Documentation/Index.rst`
-------------------------------------------

> The documentation index file at :file:`Documentation/Index.rst` is the starting

Reviewer: Maybe better would be: "The documentation start page is the entry point". "Index file" sounds pretty technical.

point of the main documentation. It usually contains general information about
the manual, a summary of its purpose and a table of contents that refers to
further pages.

> The startpage should contain an anchor target :rst:`..  _start:` above its

Reviewer: ```suggestion
The start page should contain an anchor target :rst:`..  _start:` above its
```

    See :ref:`TYPO3 Explained <t3coreapi:start>`.

> If your manual has more pages then this one it must contain a table of content

Reviewer: ```suggestion
If your manual has more pages then this start page it must contain a table of content
```

More clear.

directive called :rst:`..  toctree::`. The `toctree` on the start page
defines which pages will be displayed in main navigation of the rendered
manual.

.. _index-rst-table-of-contents:

Configure the menu - the toctree
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The `toctree` on the startpage can be hidden, it then only influences the main

Reviewer: ```suggestion
The `toctree` on the start page can be hidden, it then only influences the main
```

        Usage/Index
        Contribution/Index

For large pages it is advisable to limit the number of menu levels that will
be displayed on the startpage by setting :rest:`:maxdepth:`. This will not
limit the level of pages displayed in the main navigation.

For more information on possible options on the `toctree` directive see
chapter :ref:`Toctrees in the reST reference <toctree>`.

It is possible to use more then one `toctree` directive on the start page.
Each `toctree` should have a :rst:`:caption:` in this case.

The main navigation will then be divided into sections labeled by those
captions. The menu of the manual you are currently reading is an example for
that:

..  code-block:: rst
    :caption: Documentation/Index.rst

    ..  toctree::
        :hidden:

>         :caption: BASICS

Reviewer: I would avoid uppercase words. If they should be uppercase by design, that should be done by CSS. If not, it is up to the user and we should not give that example.

>         :caption: BASICS

Author: ```suggestion
        :caption: Basics
```

        About
        HowToGetHelp
        ...

    .. toctree::
        :hidden:

>         :caption: HOWTOS

Reviewer: ```suggestion
        :caption: Howtos
```

        WritingContent/Index
        WritingDocForExtension/Index
        ...

    ..  toctree::
        :hidden:

>         :caption: ADVANCED

Reviewer: ```suggestion
        :caption: Advanced
```

        HowToAddTranslation/Index
        GeneralConventions/HowToUpdateDocs
        GeneralConventions/ReviewInformation

    ..  toctree::
        :hidden:

>         :caption: MAINTAINERS

Reviewer: ```suggestion
        :caption: Maintainers
```
